Stop Octoprint from logging errors during startup

w1ll1am23 opened the pull request:
**Description:**
Switched to discovery.load_platform however the following Traceback is now occurring. Even though this Traceback is thrown all of the sensors still get setup. @balloob 

```bash
Traceback (most recent call last):
  File "/home/pi/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.32.0.dev0-py3.4.egg/homeassistant/helpers/entity_component.py", line 148, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/pi/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.32.0.dev0-py3.4.egg/homeassistant/components/sensor/octoprint.py", line 49, in setup_platform
    for octo_type in monitored_conditions:
TypeError: 'NoneType' object is not iterable
```

**Checklist:**

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

Commit:
Fix log errors

balloob commented:
You'll need to delete the constants that you're no longer using: DISCOVER_BINARY_SENSORS, DISCOVER_SENSORS.

I'm surprised it was even working before as I can't see those constants to be defined?

balloob commented:
The exception makes sense. The platform is now setup with `discovery_info` which you set to an empty dict. Passing in the config is only used for setting up the component if it was not set up.

Commit:
Remove discovery code

w1ll1am23 commented:
@balloob I figured it out, when I added octoprint it was never designed to to work with discovery, you have to manually add which sensors you want to the config. So... I am assuming the way it works was discovery was trying to setup the components and failed, but then the hard coded sensors in the config got hit and setup successfully? I removed discovery altogether and now it is working without any errors. I think it would probably be possible to make discovery work, but since that isn't how it was originally developed I think that should be an enhancement at a later date.

balloob commented:
OK 🐬
